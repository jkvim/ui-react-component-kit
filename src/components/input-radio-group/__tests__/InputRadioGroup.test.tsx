import * as React from 'react';
import { mount } from 'enzyme';
import { InputRadioGroup } from '../';
import toJSON from 'enzyme-to-json';

describe('InputRadioGroup', () => {
  let wrapper;

  const callback = jest.fn();
  const options = [{label: 'one', value: '1'}, {label: 'two', value: '2'}];

  beforeEach(() => {
    wrapper = mount(
      <InputRadioGroup
        id="testId"
        name="testName"
        options={options}
        onChange={callback}
        value={'2'}
      />
    );
  });

  it('should match snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render radios', () => {
    const inputs = wrapper.find('input');

    expect(inputs.get(0).props.checked).toBe(false);
    expect(inputs.get(1).props.checked).toBe(true);
  });

  it('should change radio', () => {
    const input = wrapper.find('input').at(0).simulate('change');
    expect(input.props().value).toBe('1');
  });

});
