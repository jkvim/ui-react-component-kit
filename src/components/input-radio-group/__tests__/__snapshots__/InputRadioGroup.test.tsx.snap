// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadioGroup should match snapshot 1`] = `
<InputRadioGroup
  id="testId"
  name="testName"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "one",
        "value": "1",
      },
      Object {
        "label": "two",
        "value": "2",
      },
    ]
  }
  value="2"
>
  <styled.div>
    <div
      className="sc-gZMcBi gxaaKf"
    >
      <styled.div
        alignment="horizontal"
      >
        <div
          className="sc-gqjmRU etyENR"
        >
          <styled.div
            alignment="horizontal"
            key="0"
          >
            <div
              className="sc-VigVT bezPYI"
            >
              <styled.input
                checked={false}
                hasError={false}
                id="testId-0"
                innerRef={[Function]}
                name="testName"
                onChange={[Function]}
                theme={
                  Object {
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#178acc",
                      "checked": "#2fa4e7",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "hover": "#178acc",
                    },
                  }
                }
                type="radio"
                value="1"
              >
                <input
                  checked={false}
                  className="sc-EHOje iLPXhd"
                  id="testId-0"
                  name="testName"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
              </styled.input>
              <styled.label
                hasError={false}
                htmlFor="testId-0"
                theme={
                  Object {
                    "color": Object {
                      "border": "#2fa4e7",
                      "error": "#c71c22",
                      "font": "#868e96",
                    },
                  }
                }
              >
                <label
                  className="sc-htpNat ftmpvS"
                  htmlFor="testId-0"
                >
                  one
                </label>
              </styled.label>
            </div>
          </styled.div>
          <styled.div
            alignment="horizontal"
            key="1"
          >
            <div
              className="sc-VigVT bezPYI"
            >
              <styled.input
                checked={true}
                hasError={false}
                id="testId-1"
                innerRef={[Function]}
                name="testName"
                onChange={[Function]}
                theme={
                  Object {
                    "color": Object {
                      "background": "#ffffff",
                      "border": "#178acc",
                      "checked": "#2fa4e7",
                      "default": "#2fa4e7",
                      "disabled": "#acacac",
                      "error": "#c71c22",
                      "focus": "#54b4eb",
                      "hover": "#178acc",
                    },
                  }
                }
                type="radio"
                value="2"
              >
                <input
                  checked={true}
                  className="sc-EHOje iLPXhd"
                  id="testId-1"
                  name="testName"
                  onChange={[Function]}
                  type="radio"
                  value="2"
                />
              </styled.input>
              <styled.label
                hasError={false}
                htmlFor="testId-1"
                theme={
                  Object {
                    "color": Object {
                      "border": "#2fa4e7",
                      "error": "#c71c22",
                      "font": "#868e96",
                    },
                  }
                }
              >
                <label
                  className="sc-htpNat ftmpvS"
                  htmlFor="testId-1"
                >
                  two
                </label>
              </styled.label>
            </div>
          </styled.div>
        </div>
      </styled.div>
      <styled.div
        hasError={false}
        theme={
          Object {
            "color": Object {
              "error": "#c71c22",
            },
          }
        }
      >
        <div
          className="sc-cSHVUG dYbvrH"
        />
      </styled.div>
    </div>
  </styled.div>
</InputRadioGroup>
`;
